executor/runtime/docker/docker.go

```diff
@@ -2055,9 +2055,14 @@ func (r *DockerRuntime) getContainerSharedVolumeSourceMounts(mainContainerRoot s
 			if v.FlexVolume.Options["sourceContainer"] != "main" && v.FlexVolume.Options["sourceContainer"] != "" {
 				return nil, fmt.Errorf("only 'main' SharedContainerVolume volumes are supported. Volume: %+v", v)
 			}
+			source := filepath.Join(mainContainerRoot, v.FlexVolume.Options["sourcePath"])
+			err := mkdirAllInContainer(source)
```

Comment:
Do we also need an `mkdirAllInContainer` to create `volumeMount.MountPath` in the container that will have the mount?

Reply:
Shouldn't we be able to mkdir in the tarball process that we use to create /titus and such?

Reply:
I assumed that the mountPath in the container itself would already exist, as that is a path controlled by the person doing the thing? But you are right maybe that is not the case.

@sargun Yes in theory that is possible, I didn't want to overload that tar function too much (seems weird to me), because here we can get more direct about what we are doing, and print better errors. In the tar function there is no such thing as "mkdir -p if a folder doesn't exist"

Reply:
If you mkdir in that code, and the path already exists (as long as there's no conflict with directories vs. files), it should "just work" (in terms of mkdir -p style).

Reply:
No, we can't assume the mountPath exists in the image. For example, the newtagen sidecar needs to be able to mount arbitrary paths from the main container to enable file sync.

Reply:
docker will create the destination automatically for you if it doesn't exist.
```docker run -it --mount type=bind,source="$(pwd)",target=/doesntexist busybox ls /doesntexist```